server/controllers/boardController.js

```diff
@@ -47,7 +47,48 @@ const createBoard = asyncHandler(async (request, response) => {
 // @route   PUT /api/workspaces/:workspaceId/boards/:boardId
 // @access  Private
 const editBoard = asyncHandler(async (request, response) => {
-  // To-do
+  const { name, description } = request.body;
+  const workspaceId = request.params.workspaceId;
+  const boardId = request.params.boardId;
+  const currentUserId = request.user._id;
+
+  // Find the workspace
+  const workspace = await Workspace.findById(workspaceId);
+
+  if (!workspace) {
+    response.status(404);
+    throw new Error("Workspace not found");
+  }
+
+  // Check if the user has access to the workspace
+  if (
+    workspace.creator.toString() !== currentUserId.toString() &&
+    !workspace.members.includes(currentUserId)
+  ) {
+    response.status(403);
+    throw new Error("Unauthorized access to this workspace");
+  }
+
+  // Find the board to edit
+  const boardToEdit = workspace.boards.id(boardId);
+
+  if (!boardToEdit) {
+    response.status(404);
+    throw new Error("Board not found");
+  }
+
+  // Update the board's information
+  boardToEdit.name = name || boardToEdit.name;
+  boardToEdit.description = description || boardToEdit.description;
+
```

Comment:
We need to add a check if the new board name already exists in the list of the workspace's boards. If so, throw an error.

Reply:
Good point. Updated now:

![image](https://github.com/Dining-Philosophers-Tweello/tweello/assets/70112397/7910b8ab-fc6f-4acc-b598-b9f6a115c464)